api/nodes/serializers.py

```diff
@@ -437,10 +437,14 @@ def create(self, validated_data):
                     'visible': contributor.visible
                 })
                 if not contributor.user.is_registered:
-                    node.add_unregistered_contributor(
-                        fullname=contributor.user.fullname, email=contributor.user.email, auth=auth,
-                        permissions=parent.get_permissions(contributor.user)
-                    )
+                    if contributor.user.has_usable_username():
+                        node.add_unregistered_contributor(
+                            fullname=contributor.user.fullname, email=contributor.user.email, auth=auth,
+                            permissions=parent.get_permissions(contributor.user)
+                        )
+                    elif node._primary_key not in contributor.user.unclaimed_records:
```

Comment:
Seems like this logic would be better handled in `user.add_unclaimed_record` but it might be a little late in the game to be changing model functionality. @icereval thoughts?